core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2014,8 +2014,10 @@ private[spark] object Utils extends Logging {
       } catch {
         case e: Exception if isBindCollision(e) =>
           if (offset >= maxRetries) {
-            val exceptionMessage =
-              s"${e.getMessage}: Service$serviceString failed after $maxRetries retries!"
+            val exceptionMessage = s"${e.getMessage}: Service$serviceString failed after " +
+              s"$maxRetries retries! Consider explicitly setting the appropriate port for" +
+              s"$serviceString (for example spark.ui.port for SparkUI) to an available port " +
+              "or increasing spark.port.maxRetries."
```

Comment:
worth also mentioning that choosing a port of `0` has the magic property of finding an open port?